libs/causality/src/lib/CausalAnalysisDashboard/Controls/CausalAnalysisView/CausalIndividualView/CausalIndividualChart.tsx

```diff
@@ -458,6 +458,40 @@ export class CausalIndividualChart extends React.PureComponent<
         trace.y = rawY;
       }
     }
+    if (
+      this.context.jointDataset.datasetMetaData?.featureMetaData !== undefined
+    ) {
+      const identityFeatureName =
+        this.context.jointDataset.datasetMetaData.featureMetaData
+          ?.identity_feature_name;
+
+      if (identityFeatureName !== undefined) {
+        const jointDatasetFeatureName =
+          this.context.jointDataset.getJointDatasetFeatureName(
+            identityFeatureName
+          );
+
+        if (jointDatasetFeatureName !== undefined) {
+          const metaIdentityFeature =
+            this.context.jointDataset.metaDict[jointDatasetFeatureName];
+          const rawIdentityFeature = JointDataset.unwrap(
+            dictionary,
+            jointDatasetFeatureName
+          );
+          hovertemplate += `${localization.Common.identityFeature}(${metaIdentityFeature.label}): {point.customdata.ID}<br>`;
+          rawIdentityFeature.forEach((val, index) => {
+            if (metaIdentityFeature?.treatAsCategorical) {
+              customdata[index].ID =
+                metaIdentityFeature.sortedCategoricalValues?.[val];
+            } else {
+              customdata[index].ID = (val as number).toLocaleString(undefined, {
+                maximumSignificantDigits: 5
```

Comment:
nit: can the 5 be moved to a constant, perhaps at top of file or a constants file?  Is the magic number re-used anywhere?

Reply:
There seems to be a lot of places where the number of significant digits is set to either three or five. 
![image](https://user-images.githubusercontent.com/47334368/184726964-e8ff03b9-d6cf-4e6c-b7e1-590711fe292d.png). Maybe we need to see which value to keep before declaring it as a constant.